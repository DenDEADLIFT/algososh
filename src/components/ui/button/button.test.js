import { Button } from './button'
import renderer from 'react-test-renderer'
import { render, screen, fireEvent } from "@testing-library/react"

describe('Тест компонента Button', () => {
    it('Без текста', () => {
        const tree = renderer.create(<Button />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('С текстом', () => {
        const tree = renderer.create(<Button text='Тест компонента Button' />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('В состоянии disabled', () => {
        const tree = renderer.create(<Button disabled={true} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('С лоадером', () => {
        const tree = renderer.create(<Button isLoader={true} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Вызывает alert', () => {
        const click = jest.fn()

        render(<Button onClick={click} text='Вызывает aler' />)

        const link = screen.getByText('Вызывает aler')

        fireEvent.click(link)

        expect(click).toHaveBeenCalled()
    })
})