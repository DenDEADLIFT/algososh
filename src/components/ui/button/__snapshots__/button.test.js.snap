// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Button Без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест компонента Button В состоянии disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест компонента Button С лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тест компонента Button С текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Тест компонента Button
  </p>
</button>
`;
